ios/RNCCameraRollManager.m

```diff
@@ -215,96 +215,108 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
-                 @"latitude": @(loc.coordinate.latitude),
-                 @"longitude": @(loc.coordinate.longitude),
-                 @"altitude": @(loc.altitude),
-                 @"heading": @(loc.course),
-                 @"speed": @(loc.speed), // speed in m/s
-               } : @{})
-             }
-        }];
+      // If we've accumulated enough results to resolve a single promise
+      if (first == assets.count) {
+        *stopAssets = YES;
+        stopCollections_ = YES;
+        hasNextPage = YES;
+        RCTAssert(resolvedPromise == NO, @"Resolved the promise before we finished processing the results.");
+        RCTResolvePromise(resolve, assets, hasNextPage);
+        resolvedPromise = YES;
+        return;
+      }
+
+      NSString *const assetMediaTypeLabel = (asset.mediaType == PHAssetMediaTypeVideo
+                                            ? @"video"
+                                            : (asset.mediaType == PHAssetMediaTypeImage
+                                                ? @"image"
+                                                : (asset.mediaType == PHAssetMediaTypeAudio
+                                                  ? @"audio"
+                                                  : @"unknown")));
+      CLLocation *const loc = asset.location;
+
+      // A note on isStored: in the previous code that used ALAssets, isStored
+      // was always set to YES, probably because iCloud-synced images were never returned (?).
+      // To get the "isStored" information and filename, we would need to actually request the
+      // image data from the image manager. Those operations could get really expensive and
+      // would definitely utilize the disk too much.
+      // Thus, this field is actually not reliable.
+      // Note that Android also does not return the `isStored` field at all.
+      [assets addObject:@{
+        @"node": @{
+          @"type": assetMediaTypeLabel, // TODO: switch to mimeType?
+          @"group_name": currentCollectionName,
+          @"image": @{
+              @"uri": uri,
+              @"height": @([asset pixelHeight]),
+              @"width": @([asset pixelWidth]),
+              @"isStored": @YES, // this field doesn't seem to exist on android
+              @"playableDuration": @([asset duration]) // fractional seconds
+          },
+          @"timestamp": @(asset.creationDate.timeIntervalSince1970),
+          @"location": (loc ? @{
+              @"latitude": @(loc.coordinate.latitude),
+              @"longitude": @(loc.coordinate.longitude),
+              @"altitude": @(loc.altitude),
+              @"heading": @(loc.course),
+              @"speed": @(loc.speed), // speed in m/s
+            } : @{})
+          }
       }];
-    }];
+    };
+
+    if ([groupTypes isEqualToString:@"all"]) {
+      PHFetchResult <PHAsset *> *const assetFetchResult = [PHAsset fetchAssetsWithOptions: assetFetchOptions];
+      currentCollectionName = @"";
```

Comment:
Should this be `@"all"` instead? Not sure, but setting `@"group_name"` to an empty string could be a problem?

Reply:
The only reasonable option I can see is hardcoding "All Photos" to maintain backwards-compatibility with how CameraRoll used to work on iOS with ALAssetLibrary, and I'm not sure I love that either.

Reply:
I'm honestly not quite sure here. It's the equivalent to the album name. "All Photos" is possible, but I'm not sure what used to be emitted in that field in this situation.
I can change it to whatever you think is the best option here.

Reply:
I changed it to `"All Photos"`

Reply:
It would have been "All Photos" in the past (though I'm not sure if this was a localized string or not -- but based on some previous investigation, I don't think it's localized).